massa-factory-worker/src/block_factory.rs

```diff
@@ -192,7 +193,42 @@ impl BlockFactoryWorker {
         block_storage.extend(endo_storage);
 
         // gather operations and compute global operations hash
-        let (op_ids, op_storage) = self.channels.pool.get_block_operations(&slot);
+        let (op_ids_, op_storage) = self.channels.pool.get_block_operations(&slot);
+
+        // 'Steal' op denunciation
+        // TODO: add doc about why we stole those
+        let ops = op_storage.read_operations();
+        let mut new_ops: Vec<WrappedOperation> = Vec::new();
+        let mut ops_to_clean: PreHashSet<OperationId> = Default::default();
+        let op_ids: Vec<OperationId> = op_ids_
```

Comment:
there is a problem with theft: it changes the operation ID so the list of previously executed op IDs is desync

Reply:
we can discuss this